src/Http/Http.Extensions/src/HttpMetricsEndpointConventionBuilderExtensions.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Builder;
+
+/// <summary>
+/// HTTP metrics extension methods for <see cref="IEndpointConventionBuilder"/>.
+/// </summary>
+public static class HttpMetricsEndpointConventionBuilderExtensions
+{
+    /// <summary>
+    /// Specifies that HTTP request duration metrics is disabled for an endpoint.
+    /// </summary>
+    /// <typeparam name="TBuilder">The type of endpoint convention builder.</typeparam>
+    /// <param name="builder">The endpoint convention builder.</param>
+    /// <returns>The original convention builder parameter.</returns>
+    public static TBuilder DisableHttpMetrics<TBuilder>(this TBuilder builder) where TBuilder : IEndpointConventionBuilder
+    {
+        builder.Add(b => b.Metadata.Add(new DisableHttpMetricsAttribute()));
```

Comment:
In reference to whether or not this metadata should be stored or reused. We _do_ do this for the `ExcludeFromDescription` metadata used for OpenAPI ([ref](https://github.com/dotnet/aspnetcore/blob/71b5ef3f5c997fbbb9ff3dc1a16d58e8ccdeed52/src/Http/Routing/src/Builder/OpenApiRouteHandlerBuilderExtensions.cs#L17)).

I have a slight preference to the stored/reused pattern because it more clearly indicates that the metadata is stateless.